tests/config/test_config.py

```diff
@@ -169,40 +165,32 @@ def test_default_config(self):
         conf._validate_all()
 
 
-class ConfigInitTests(unittest.TestCase):
+class TestConfigInit:
 
     """Test initializing of the config."""
 
-    def setUp(self):
-        self.temp_dir = tempfile.mkdtemp()
-        self.conf_path = os.path.join(self.temp_dir, 'config')
-        self.data_path = os.path.join(self.temp_dir, 'data')
-        self.cache_path = os.path.join(self.temp_dir, 'cache')
-        os.mkdir(self.conf_path)
-        os.mkdir(self.data_path)
-        os.mkdir(self.cache_path)
+    @pytest.yield_fixture(autouse=True)
+    def setup(self, tmpdir):
+        self.conf_path = (tmpdir / 'config').ensure(dir=1)
+        self.data_path = (tmpdir / 'data').ensure(dir=1)
+        self.cache_path = (tmpdir / 'cache').ensure(dir=1)
         self.env = {
-            'XDG_CONFIG_HOME': self.conf_path,
-            'XDG_DATA_HOME': self.data_path,
-            'XDG_CACHE_HOME': self.cache_path,
+            'XDG_CONFIG_HOME': str(self.conf_path),
```

Comment:
I'm wondering if we are running into problems with the ``str`` cast as Python 3 uses unicodes by default :worried: